apis/discoveryengine/v1alpha1/targetsite_types.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DiscoveryEngineDataStoreTargetSiteGVK = GroupVersion.WithKind("DiscoveryEngineDataStoreTargetSite")
+
+// DiscoveryEngineDataStoreTargetSiteSpec defines the desired state of DiscoveryEngineDataStoreTargetSite
+// +kcc:proto=google.cloud.discoveryengine.v1.TargetSite
+type DiscoveryEngineDataStoreTargetSiteSpec struct {
+	// The DataStore this target site should be part of.
+	DataStoreRef *DiscoveryEngineDataStoreRef `json:"dataStoreRef,omitempty"`
+
+	// The resource ID is server-generated, so no ResourceID field
+
+	// Required. Input only. The user provided URI pattern from which the
+	// `generated_uri_pattern` is generated.
+	ProvidedUriPattern *string `json:"providedUriPattern,omitempty"`
```

Comment:
nit: Uri --> URI

Reply:
Do you mind adding this acronym to the CRD validation list? https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/apichecks/crds_test.go#L195